homeassistant/components/yolink/__init__.py

```diff
@@ -80,6 +92,28 @@ def on_message_callback(message: tuple[str, BRDP]) -> None:
         device_coordinator = device_coordinators.get(device_id)
         if device_coordinator is None:
             return
+        if device_coordinator.device.device_type == ATTR_DEVICE_SMART_REMOTE:
+            device_registry = dr.async_get(hass)
+            device_entry = device_registry.async_get_device(
+                identifiers={(DOMAIN, device_coordinator.device.device_id)}
+            )
+            if device_entry is not None:
+                btn_press_event = resolved_state.get("event")
+                if btn_press_event is not None:
+                    key_mask = btn_press_event["keyMask"]
+                    btn_no = int(math.log2(key_mask)) + 1
```

Comment:
The library needs to be improved to create a better interface for the events. The library user should not need to parse the events like this, eg splitting strings, accessing dict by strings, and doing log math.

Please build a nice api in the library and return typed instances where different values can be accessed by object attributes. I recommend using pydantic to deserialize api data.